### PR TITLE
05 single linked list tests

### PR DESCRIPTION
**Pull request by PaulFrmBrn**

### CODE REVIEW

File: src/main/java/linked/list/LinkedList.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
+package linked.list;
+
+import java.util.NoSuchElementException;
+import java.util.function.Consumer;
+
+/**
+ * Linked List interface
+ *
+ * Describes method fot one directional Linked List
```
Comment (zamonier) [inline src/main/java/linked/list/LinkedList.java:9]: nit: `method fot` -> `methods for`

File: src/main/java/linked/list/LinkedList.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ *
+ * @author Dmitry Pavlov
+ * @since 19.04.2020
+ */
+public interface LinkedList<T> {
+
+    /**
+     * Returns the size of the list
+     */
+    int getSize();
```
Comment (zamonier) [inline src/main/java/linked/list/LinkedList.java:31]: just `size()` more familiar for me

File: src/main/java/linked/list/LinkedList.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    /**
+     * Inserts element at the specified location of the list
+     * @throws IndexOutOfBoundsException if location is not in [0, size) range
+     * @throws IllegalArgumentException if value is null
+     */
+    void insertAt(int location, T value);
+
+    /**
+     * Returns location of the element with specified value
+     * @throws NoSuchElementException if there is on element with specified value in the list
```
Comment (zamonier) [inline src/main/java/linked/list/LinkedList.java:54]: nit: `on` -> `no`

File: src/main/java/linked/list/LinkedList.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+     * @throws IllegalArgumentException if value is null
+     */
+    int search(T value);
+
+    /**
+     * Applies specified function to the each and every element of th elist
```
Comment (zamonier) [inline src/main/java/linked/list/LinkedList.java:60]: nit: `th elist` -> `the list`

File: src/main/java/linked/list/LinkedList.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    /**
+     * Deletes element at the specified location of the list
+     * @throws IllegalArgumentException if there are no element in the list
+     * @throws IndexOutOfBoundsException if location is not in [0, size) range
+     */
+    void deleteAt(int location);
+
+    /**
+     * Deletes all element from the list
```
Comment (zamonier) [inline src/main/java/linked/list/LinkedList.java:84]: nit: `element` -> `elements`

File: src/test/java/linked/list/single/SingleLinkedListTest.java [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+ * @since 19.04.2020
+ */
+class SingleLinkedListTest {
+
+    @Test
+    public void shouldCreateEmptyList() {
+        // when
+        var list = new SingleLinkedList<Integer>();
+        // then
+        assertEquals(list.getSize(), 0);
```
Comment (zamonier) [inline src/test/java/linked/list/single/SingleLinkedListTest.java:27]: shouldCreateEmptyList? - it's enough to check not null I think

File: src/main/java/linked/list/LinkedList.java (hunk 1 above)
Comment (PaulFrmBrn) [reply src/main/java/linked/list/LinkedList.java:9]: thanks, will fix

File: src/main/java/linked/list/LinkedList.java (hunk 2 above)
Comment (PaulFrmBrn) [reply src/main/java/linked/list/LinkedList.java:31]: ok, will fix

File: src/main/java/linked/list/LinkedList.java (hunk 3 above)
Comment (PaulFrmBrn) [reply src/main/java/linked/list/LinkedList.java:54]: thanks, will fix

File: src/main/java/linked/list/LinkedList.java (hunk 4 above)
Comment (PaulFrmBrn) [reply src/main/java/linked/list/LinkedList.java:60]: thanks, will fix

File: src/main/java/linked/list/LinkedList.java (hunk 5 above)
Comment (PaulFrmBrn) [reply src/main/java/linked/list/LinkedList.java:84]: thanks, will fix

File: src/test/java/linked/list/single/SingleLinkedListTest.java (hunk 6 above)
Comment (PaulFrmBrn) [reply src/test/java/linked/list/single/SingleLinkedListTest.java:27]: Checking size is not only about  nullability but initialisation of the `size` value.

File: src/main/java/linked/list/LinkedList.java (hunk 1 above)
Comment (PaulFrmBrn) [reply src/main/java/linked/list/LinkedList.java]: done

File: src/main/java/linked/list/LinkedList.java (hunk 2 above)
Comment (PaulFrmBrn) [reply src/main/java/linked/list/LinkedList.java]: done

File: src/main/java/linked/list/LinkedList.java (hunk 5 above)
Comment (PaulFrmBrn) [reply src/main/java/linked/list/LinkedList.java]: done

File: src/main/java/linked/list/LinkedList.java (hunk 4 above)
Comment (PaulFrmBrn) [reply src/main/java/linked/list/LinkedList.java]: done

**Pull request merged**